source/common/common/logger.h

```diff
@@ -430,6 +431,22 @@ template <Id id> class Loggable {
     }                                                                                              \
   } while (0)
 
+// This is to get us to pass the format check. We reference a real-world time source here.
+// I think it's not easy to mock/simulate time here as it stands today anyway.
+using t_logclock = std::chrono::steady_clock; // NOLINT
+
+#define ENVOY_LOG_PERIODIC(LEVEL, CHRONO_DURATION, ...)                                            \
+  do {                                                                                             \
+    static auto* last_hit = new std::atomic<uint64_t>();                                           \
```

Comment:
I'd strongly advise checking `LEVEL` here first to remove unnecessary code and memory barriers for messages that will get discarded by `ENVOY_LOG_COMP_LEVEL` later anyway.
Memory barriers also introduce optimization barriers (this code won't get optimized away) which can result in extra memory accesses, slightly bigger code size and possibly in some dead code being left in the resulting binary there (dead code can be a security concern).

Reply:
@glodomorski thanks! that makes sense. I addressed your feedback in 85003d4.